### ISSUE TITLE
Add instances review functionality

### ISSUE THREAD

**Issue by lakshyagupta21** [opendatakit/share#39]
Add a functionality to review the forms received from other devices and give feedback on received forms.

**Comment by lakshyagupta21** [comment]
@shobhitagarwal1612 @lognaturel Need some ideas on this like how the review process should look like.

One way that I can think of is open forms in edit mode in collect from share app and when a user finishes it and navigates back to the share app we add an option to give feedback on the same form. The only problem that I'm able to see in this method is that when a user navigates to collect to edit/view the form, we would not be able to know that user actually viewed/edited the form.

**Comment by shobhitagarwal1612** [comment]
Instance review status: Pending, Approved, Rejected

Pending - yet to be reviewed
Approved - Will be moved to collect's instances dir
Rejected - To be resent back to the enumerator

1. Go to `Review forms section` (this will contain the list of received instances which are tagged as `Pending`
2. Click on any instance, this will open the instance in Collect in `View only` mode
3. After going through the instance, the user presses back and returns to `Share` app
4. The user is greeted with a feedback screen with options like, `approve` :+1: , `reject` :-1: , `review later` :clock10: and a textbox to write feedback

Tagging everyone to give your feedback.
@lakshyagupta21 @lognaturel @yanokwa @smoyth @aurdipas

**Comment by lakshyagupta21** [comment]
Currently, whenever a form is received, it is added to collect's instances dir, the reason for doing it this way is that sometimes a user may don't want to get any feedback from the supervisor, there might be a possibility that for a single form there are multiple enumerators who fill different sections of the form.

**Comment by lakshyagupta21** [comment]
Any suggestions?

![review](https://user-images.githubusercontent.com/18495162/42421401-a64af6e0-82f2-11e8-89ba-4b86c0bccc06.png)

**Comment by lognaturel** [comment]
> we would not be able to know that user actually viewed/edited the form.

I think this is ok. We sort of need to assume that supervisors can be trusted. That said, you could keep a counter of how many times the button to open the form in Collect was tapped, right? That might help a supervisor remember whether they have looked at it or not.

> whenever a form is received, it is added to collect's instances dir, the reason for doing it this way is that sometimes a user may don't want to get any feedback from the supervisor, there might be a possibility that for a single form there are multiple enumerators who fill different sections of the form.

I very much agree with this approach.

Thanks for providing something concrete to try and comment on in #41. Some thoughts:
- "View form again" opens Collect, right? Could this say "View in ODK Collect"? to be a bit more specific?
- What are you thinking the "Review later" button will do? Keep it as pending and go to the next filled form?
- It might be helpful to show more information on which filled form is being displayed. For example, you could get the `instanceName` from Collect and display that on the review screen. The instance name can be configured in the form (http://xlsform.org/#instance-name) to be something like "All widgets - Sally Johnson - July 3rd, 2018".

**Comment by lakshyagupta21** [comment]
> Could this say "View in ODK Collect"? to be a bit more specific?

Seems fine to me, this will give some idea to the user that this app needs Collect to view the form.



> What are you thinking the "Review later" button will do?

It will mark the form as pending and move back to the tabs screen.




> It might be helpful to show more information on which filled form is being displayed. For example, you could get the instanceName from Collect and display that on the review screen.

Currently it is showing `displayName` on the screen.

**Comment by lognaturel** [comment]
> Currently it is showing displayName on the screen.

That's the display name of the form, right? That means every single filled form from the same form will look exactly alike?

**Comment by lakshyagupta21** [comment]
> That's the display name of the form, right?

Yes, in the snapshot it is showing `All-widgets` which created some confusion as it is also a displayName of a `all-widgets` form. 


> That means every single filled form from the same form will look exactly alike?

Yes

**Comment by lognaturel** [comment]
Ahh, and you don't have `instanceName` available because it was only in the database on the sender side, not in the form, right? Do you know what device it was sent from? Maybe you can start by showing that and the date/time it was received? We can file this as a separate issue but eventually you could parse the instance to compute the `instanceName` and display that. Alternately you could pull out a fixed field like `username`.

> when a user navigates to collect to edit/view the form, we would not be able to know that user actually viewed/edited the form.

How about storing and displaying the number of times the "View form in ODK Collect" button was tapped? That would at least give the reviewer a sense of whether they've looked at this instance before.

**Comment by lakshyagupta21** [comment]
> you don't have instanceName available because it was only in the database on the sender side, not in the instance XML, right?

Do we store `instanceName` in the database I thought `displayName` is the only field which helps user to identify the form.



> Do you know what device it was sent from?

No, currently we are not tracking the device info.

> Maybe you can start by showing that and the date/time it was received?

Yes, this can be done.



> How about storing and displaying the number of times the "View form in ODK Collect" button was tapped?

How about some boolean value or int value like 0 and 1 which tells that whether user visited it before or not?

**Comment by lognaturel** [comment]
🤦‍♀️ We're talking about the same thing -- the database column is indeed called `displayName`. I only tried it with forms that don't have an expression for instance name so I didn't see that it was doing the right thing. Sorry!

> How about some boolean value or int value like 0 and 1 which tells that whether user visited it before or not?

I was thinking of a count because it might help me remember something like "oh, that's the one I really wasn't sure about" but I can see how visited/not visited is much easier from a UI standpoint. How about storing a count that gets updated but you can display it as visited/not visited for now and see what users think.

**Comment by lakshyagupta21** [comment]
> How about storing a count that gets updated but you can display it as visited/not visited for now and see what users think.

Sounds good, I'll add a new column in `transfers` table for keeping track of the number of times a user viewed the form.
